### DISCUSSION TITLE
[Projects Beta] Allow to hide "status" columns in the Board View

### DISCUSSION THREAD

**Discussion by serpro69** [community/community#6906] [Projects and Issues] [answered]
Is it possible to hide one or more "status" columns in the Board View
For example I want to hide "ToDo" column in one of the views and only display "In Progress" and "Done"

**Comment by directionless** [comment]
Same! I have a status for `triage` and I don't want to display that column on the kanban boards

**Comment by merriemcgaw** [comment]
Please! I really need this feature in my projects!

**Comment by jeremyharisch** [comment]
Our team also needs such feature - To have multiple board views one for our devs, one for our PO, TechnicalWrites, etc.
Because not everyone needs to focus on all status

**Comment by danil-topchiy** [comment]
Agree, also would love this feature!

**Comment by tubbo** [comment]
plus one-ing this feature...would be nice if blocked/on hold tickets were not on the board view so they could be triaged separately.

**Comment by LivioMirandaJr** [comment]
This is necessary in so many types of projects that I can't even understand how this isn't a thing yet.

**Comment by pmillerlokion** [comment]
Also needed by our team. Agreed too with @LivioMirandaJr that it seems essential.

**Comment by MatthewOwens** [comment]
:+1:

**Comment by pehaada** [comment]
+1

**Comment by paziz001** [comment]
👍

**Comment by NuclearRedeye** [comment]
Surprised to find this didn't work, definitely a 'must have' feature as otherwise the ability to have multiple boards/views seems rather limited.

Being specific, what I'd like to see is the ability to show/hide specific status columns for a specific board . A 'clunky' implementation might be to extend the filter logic so that when a specific status is filtered out the column is hidden. This seems to work for issues with 'No Status' where the column/pipeline hides itself when I apply the filter `-no:status` but it doesn't work for others.

I personally think the ideal would be to extend the per-column menu (image below) with a toggle for show/hide which each board then remembers.

<img width="520" alt="Screenshot 2022-03-01 at 10 58 26" src="https://user-images.githubusercontent.com/52923935/156157098-51384738-1c31-4a03-9e84-6a9a2f9b412d.png">

🤞 this feature can be added and made available soon.

**Comment by otonielguajardo** [comment]
+1

**Comment by hieuh25** [comment]
+1

**Comment by andrenicodemus** [comment]
Without this feature we are limited to having all views (in our case, separate views exist for different initiatives) with the exact same status columns. That basically means we can't have, for instance, a view for our Product Owner to gather and organize tasks before they are distributed into the initiatives' kanban boards.

**Comment by malahaas** [comment]
Agreed. I'd like status columns hidden too. Would be great if it was driven by the filter. eg: "-Status:STATUS_NAME" should hide the STATUS_NAME column. Or, if you use "Status:STATUS_1,STATUS_2" as a filter, only those 2 columns should be shown.

**Comment by danielBingham** [comment]
Adding my voice to those requesting this.

Another potential approach would simply be to not show issues that don't have a column matching their status on the board.  Right now they go into an automatically created "No status" column.  The default should be to simply not show them.  This would allow us to create multiple boards with non-overlapping statuses for different parts of a development pipeline.

We could have one board for planning and design with "Under consideration", "In Planning", "In Design", and "Ready for Development".  Then we could have another board for development with "Ready", "In progress", "In Code Review", "In QA" and "Done".  Finally, issues with out any status would only show in the list view, rather than the board view, allowing the creation of a separate backlog or icebox.

It seems like a pretty simple tweak and it would be a big improvement on the usability of the new Projects!

**Reply by XGeffrier** [reply]
While I support the request to be able to hide a column, I think it is ok to keep seeing empty columns if you want to.

When you are used to seeing the same four columns every day, it can be confusing when the third one is missing just because it is empty. More simply, one may want to enjoy the sight of a delightfully empty "To-Do" column.

**Comment by ivanaguilario** [comment]
This is a basic feature that should already be implemented imo. It is extremely limiting.

**Comment by erinmez** [comment]
Hiding columns would be super useful! We use a kanban board for different teams and it would help to hide the first column (which is the "No <current team>" column in our case)

**Reply by erinmez** [reply]
We have two separate Single Select Fields named "Team Dev" & "Team Enable" and use two Kanban boards which are configured with "Team Dev" and "Team Enable" as column fields.

We are able to remove the "No Status" column by applying the filter `-no:team---dev`  in the "Team Dev" board and `-no:team---enable` in the "Team Enable" board. 

With this setup, the PM assigns issues into backlogs for each team (by selecting the value "Backlog" in the single select field) and orders the priority in the backlog column in the Team's board. 

Team members then pull issues as usual from right to left and from the top of their backlog eventually.

**Comment by nestor-sk** [comment]
`document.querySelectorAll('[data-board-column="YOUR_STATUS_NAME"]')[0].style.display = "none"` does the job. But it is sad to have to use custom JS (or CSS) tools for such a simple feature

**Reply by sherif-nedap** [reply]
what kind of tools did you use to run this automatically ?

**Reply by nestor-sk** [reply]
A browser extension like TamperMonkey or Custom JavaScript for Websites

**Comment by Nizzlay** [comment]
+1 on this.

As a workaround for now I have created multiple status fields. Mainly `Status` and `Backlog status`. So that I could create a board for PO which has the `Backlog status` statusses, seperate from a sprint board with the `Status` statusses.

**Comment by abbyachau** [comment]
+1. This feature would be super useful.

**Comment by StoneCypher** [comment]
Agreed.  I'd like a board view which suppresses `done`, `declined`, and `backlog`, so that I can focus on what's actually in flight.

**Comment by travis-jm** [comment]
+1

**Comment by xcitic** [comment]
+1

**Comment by randseay** [comment]
+1

**Comment by GregAC** [comment]
Another +1 for this simple, yet essential, feature

**Comment by platschi** [comment]
+1

**Comment by danmoore2205** [comment]
+1

**Comment by craig006** [comment]
+1

**Comment by MathiasSoender** [comment]
+1

**Comment by nbozzelli** [comment]
+1

**Comment by amfio** [comment]
As this lack of functionally was one of the features preventing my team making the transition to the new GitHub projects, I have created a Chrome [extension](https://chrome.google.com/webstore/detail/github-projects-column-hi/mmaejmnikaleejkgkcjmabmmchejjlol) which looks for the `-status` filters and removes the columns which match that filter. 

For example, without the extension: the a filter of `-status:Backlog,Blocked` would remove all cards with a status of `Backlog` or `Blocked` but the board would still keep columns for those statuses even though they are guaranteed to be empty. All this extension does is remove those columns from the board.

Feel free to give it a go and see if it works for you and submit any [issues](https://github.com/amfio/github-projects-column-hider/issues) you have.

Alternatively you could grab the [source code](https://github.com/amfio/github-projects-column-hider/blob/master/src/script.js) and dump it in something like Tampermonkey 

Hopefully GitHub will add native support for this and I will remove the extension.

**Comment by ronalddb89** [comment]
+1

**Comment by yash-lysn** [comment]
can we please STOP writing `+1`..github should block such comments!!!

**Reply by smith** [reply]
+1

**Comment by archon810** [comment, marked as answer]
Looks like Github [officially released](https://github.blog/changelog/2022-09-02-the-new-github-issues-september-2nd-update/) this today.

> 🙈 Auto-hide columns with board filters
> [You ask](https://github.com/community/community/discussions/6906), we deliver! With today’s release, project boards will automatically hide columns depending on the filters you’ve applied. Customize your boards with the exact set of columns you need – no more empty columns here!
> 
> ![188190593-ef75e9eb-e1a1-42a2-bf5d-6ec29d69ad72](https://user-images.githubusercontent.com/469543/188206227-1e931598-5719-4414-abe3-fd5fab0cd2b6.gif)

**Reply by StoneCypher** [reply]
Nice, now all we need are issue blockers

**Reply by AshleighAdams** [reply]
Does anyone know how to `or` in a filter? Trying to do `assignee:@me or no:assignee`

**Reply by rbx-max** [reply]
nice!

**Comment by nedbat** [comment]
I have a project with `-status:"Long Done"`, and it still shows a column for "Long Done".  Is there something special I have to do to trigger column hiding? Is this only going to work on newly created projects?  I also tried making a new board view in my project to show only `status:Backlog`, and it showed empty columns for all my other statuses.  :(

**Reply by 2percentsilk** [reply]
Hey @nedbat !

There shouldn't be any additional steps you need to do to enable this - however, it may be possible that your browser is stuck on a previous release of Projects. I'd try closing your browser and re-opening. Let me know if you're still seeing an error though and we'll investigate further.

**Reply by nedbat** [reply]
OK, restarting the browser fixed it.

**Comment by chicori3** [comment]
+1 ☝️
